### ISSUE TITLE
Cannot Add Node to New Cluster After Removing From a Previous Cluster

### ISSUE THREAD

**Issue by JRNC** [rancher/rancher#30167]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
Create a 3 node cluster with Rancher, then remove one the nodes, create a new cluster, and try to add the node that was removed from the original cluster to the new cluster.

**Result:**
The node shows up as active, but the cluster shows the following error:

"This cluster is currently Error; areas that interact directly with it will not be available until the API is ready.
Cluster health check failed: cluster agent is not ready"

**Other details that may be helpful:**
The environment is compromised of on-prem VMs. The Rancher server was stood up using the instructions for an HA Rancher cluster using a 3 node RKE cluster. The Rancher cluster is fronted by a layer 4 NGINX load balancer that was configured using Rancher's recommended nginx.conf template. The servers are behind a proxy for internet access and are all on the same subnet.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.2
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VM (4 vCPU and 8GB RAM)
- Kubernetes version (use `kubectl version`): v1.19.3

```
(paste the output here)
```

- Docker version (use `docker version`): 19.03.13

```
(paste the output here)
```

**Comment by JRNC** [comment]
I would like to add that I have tried fully rebuilding this node, so this is not an issue with cleaning up the node itself.

**Comment by maggieliu** [comment]
@JRNC Can you clarify the two cluster's setup (how many etcd/ctrl/worker) and the roles for each of the nodes. Can you also check the logs in the downstream agents and see if there are any errors?

**Comment by JRNC** [comment]
Original cluster had three nodes with all roles on them, the new cluster I tried with one node with all roles and two nodes with all roles. The issue appears to be unrelated to this however.

FYI for anyone receiving this error... for me at least (running RHEL 7 on nodes), the issue was with lowest priority iptables rules that were set to reject input and forward traffic after the higher priority rules that were allowing the necessary communication for the node installation. It looks like the iptables chains that were put in place when the node installation occurred were added as a lower priority than the preexisting, lowest priority reject rules for the input and forward chains. I had to remove my lowest priority reject rules, run the node installation (which occurred successfully this time) and then re-add a lower priority chain for the reject rules.

It is worth noting that I was previously able to add nodes with the same iptables rules, so something has changed with iptables chain priorities. This may be something upstream with a CNI, but if possible a bugfix needs to be applied to ensure that the chains are applied at a high enough priority to prevent this behavior. These types of lowest priority reject rules and default deny chain policies are very common and this could cause issues for many folks.
